### ISSUE TITLE
How training can continue based on saved training parameters when training is interrupted.

### ISSUE THREAD

**Issue by Worker-Hou** [ge-xing/Diff-UNet#38]
Questions above, thank you for your answer!

**Comment by Worker-Hou** [comment]
Which part of the code should be changed?

**Comment by ge-xing** [comment]
The init section of the Trainer class. When you have gotten a model, and then you can load the previous weight.

**Comment by Worker-Hou** [comment]
> The init section of the Trainer class. When you have gotten a model, and then you can load the previous weight.

Sorry, I tried to call the (load_state_dict) method in the (init) function of (Trainer) in (Trainer. But still not from the previously trained weight part to continue training. Please tell me how you modified this part of the code, thank you very much!

**Comment by ge-xing** [comment]
<img width="807" alt="image" src="https://github.com/ge-xing/Diff-UNet/assets/109211094/4562926b-27e1-4e4d-b126-c0ca2eecd381">

After you have a model, you can call the load_state_dict() function to load the previous parameters.

**Comment by Worker-Hou** [comment]
> <img alt="image" width="807" src="https://private-user-images.githubusercontent.com/109211094/344466371-4562926b-27e1-4e4d-b126-c0ca2eecd381.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3NTQ1NzcsIm5iZiI6MTcxOTc1NDI3NywicGF0aCI6Ii8xMDkyMTEwOTQvMzQ0NDY2MzcxLTQ1NjI5MjZiLTI3ZTEtNGU0ZC1iMTI2LWMwY2EyZWVjZDM4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYzMFQxMzMxMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNzU3MjUyYjE4ZWU5Mzg3MjI2NzAwM2FlY2YzNjE4NTNhYzE4OWUwN2Y2NjE1MTgwZGRiNjg1YmVkZDQzOGNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.CLCm2ohkF_IxcJx60ycoh4xmmbh0kSEo0EdXVw-0DaY">
> After you have a model, you can call the load_state_dict() function to load the previous parameters.

Thank you for your reply. I have solved the problem.
This may have been because every time I called the previous weight file, the Epoch number would start from 0 again, which made me think that it did not run successfully. However, I find that loss is the value of the previous pre-training weight, so it is successful for now!
Thanks again for your reply!
